packages/devtools_app/lib/src/screens/inspector/inspector_controller.dart

```diff
@@ -248,12 +245,12 @@ class InspectorController extends DisposableController
 
   bool get detailsSubtree => parent != null;
 
-  RemoteDiagnosticsNode get selectedDiagnostic =>
+  RemoteDiagnosticsNode? get selectedDiagnostic =>
       selectedNode.value?.diagnostic;
 
-  final ValueNotifier<int> _selectedErrorIndex = ValueNotifier<int>(null);
+  final ValueNotifier<int?> _selectedErrorIndex = ValueNotifier<int?>(null);
 
-  ValueListenable<int> get selectedErrorIndex => _selectedErrorIndex;
+  ValueListenable<int?> get selectedErrorIndex => _selectedErrorIndex;
```

Comment:
making this non-nullable allows us to get rid of several null checks and asserts elsewhere. Also, I would argue that the fact that `-1` is a standard for the non-existing-index would be a reason _for_ using it here over `null`. -1 is an intuitive pattern for the index that does not exist.

Reply:
I agree with @polina-c on this case. Using null for a missing index is better because you get language support to make sure you aren't accessing a missing index. With -1, you have to make sure you are checking for -1 in all the right places.

Reply:
null checks help us to catch unexpected non-existing-index early, and analyzer will give warnings at compile time. -1 will cause out of range exception or unexpected behavior at run time, if not caught early.
null checks seem to be beneficial in this case